src/AcmProgressTracker/AcmProgressTracker.tsx

```diff
@@ -0,0 +1,183 @@
+/* Copyright Contributors to the Open Cluster Management project */
+
+import { makeStyles } from '@material-ui/core'
+import { Button, Popover, PopoverProps, Spinner, Text, TextContent } from '@patternfly/react-core'
+import React from 'react'
+import { useViewport } from '../AcmCharts/AcmChartGroup'
+import {
+    AsleepIcon,
+    CheckCircleIcon,
+    ExclamationCircleIcon,
+    ExclamationTriangleIcon,
+    MinusCircleIcon,
+    UnknownIcon,
+    ResourcesEmptyIcon,
+} from '@patternfly/react-icons'
+
+export type AcmProgressTrackerProps = {
+    steps: ProgressTrackerStep[]
+    isCentered?: boolean
+    Title: string
+    Subtitle: string
+    isStatusPopover?: boolean
+    isStacked?: boolean
+}
+
+export type ProgressTrackerStep = {
+    active: boolean
+    statusType: StatusType
+    statusText: string | React.ReactNode
+    popover?: PopoverProps
+    statusSubtitle?: string
+}
+
+export enum StatusType {
+    'healthy' = 'healthy',
+    'danger' = 'danger',
+    'warning' = 'warning',
+    'progress' = 'progress',
+    'detached' = 'detached',
+    'pending' = 'pending',
+    'unknown' = 'unknown',
+    'sleep' = 'sleep',
+    'empty' = 'empty',
+}
+
+const useStyles = makeStyles({
+    /* istanbul ignore next */
+    parentContainer: {
+        display: 'flex',
+        paddingTop: '10px',
+    },
+    popoverParentContainer: {
+        display: 'inline-grid',
+    },
+    popoverBody: {
+        display: 'flex',
+    },
+    stepContainer: {
+        display: 'inline-flex',
+        padding: '10px 0px 10px 0px',
+    },
+    text: { width: 'max-content' },
+    divider: {
+        padding: '0px 40px 0px 40px',
+        maxWidth: '180px',
+        maxHeight: '20px',
+    },
+    stepStatus: {
+        paddingLeft: '25px',
+    },
+    container: {
+        display: 'flex',
+    },
+    icon: {
+        width: '18px', // Progress size md is 18px
+    },
+    iconMargin: {
+        margin: '3px 2px 1px 2px',
+    },
+    button: {
+        padding: 0,
+        fontSize: 'inherit',
+    },
+})
+
+const divider = (isStacked: boolean) => {
+    const classes = useStyles()
+    if (isStacked) return undefined
+    return (
+        <svg className={classes.divider}>
+            <line
+                x1="0"
+                x2="100"
+                y1="10"
+                y2="10"
+                stroke="#D2D2D2" // --pf-global--palette--black-300
+                strokeWidth="2"
+                strokeLinecap="square"
+                strokeDasharray="1, 3"
+            />
+        </svg>
+    )
+}
+
+export function AcmProgressTracker(props: AcmProgressTrackerProps) {
+    const classes = useStyles()
+    const { viewWidth } = useViewport()
+    const isStacked = props.isStacked || viewWidth < 700
+
+    return (
+        <div>
```

Comment:
Seems there are a lot of divs here; are they all necessary?  Can some be converted to `Fragment` ?

Reply:
Switched to Fragments and Gallery, as PF guidelines say to use it for responsive grid-like behavior.